cmd/snap/cmd_auto_import.go

```diff
@@ -94,15 +97,31 @@ func autoImportCandidates() ([]string, error) {
 			continue
 		}
 
+		// TODO: should the following 2 checks try to be more smart like
+		//       `snap-bootstrap initramfs-mounts` and try to find the boot disk
+		//       and determine what partitions to skip using the disks package?
+
+		// skip all initramfs mounted disks on uc20
 		mountPoint := l[4]
+		if strings.HasPrefix(mountPoint, boot.InitramfsRunMntDir) {
+			continue
+		}
+
+		// skip all seed dir mount points too, as these are bind mounts to the
+		// initramfs dirs on uc20, this can show up as
+		// /writable/system-data/var/lib/snapd/seed as well as
+		// /var/lib/snapd/seed
+		if strings.HasSuffix(mountPoint, dirs.SnapSeedDir) {
```

Comment:
It would be better to skip the mount point based on source, not target. Can we identify that somehow easily, at this stage?

Reply:
That would be a previous TODO, find a device we mounted some well known partition from (eg ubuntu-boot), and then skip mounts from that device.

Reply:
yes exactly, that's the TODO above